core/src/main/java/org/apache/carbondata/core/mutate/DeleteDeltaBlockletDetails.java

```diff
@@ -69,9 +69,11 @@ public Integer getPageId() {
     if (obj == null || !(obj instanceof DeleteDeltaBlockletDetails)) {
       return false;
     }
-
     DeleteDeltaBlockletDetails that = (DeleteDeltaBlockletDetails) obj;
-    return id.equals(that.id) && pageId == that.pageId;
+    if (null == pageId || null == that.pageId) {
+      return false;
+    }
+    return id.equals(that.id) && pageId.intValue() == that.pageId.intValue();
```

Comment:
change to `return id.equals(that.id) && (pageId.intValue() == that.pageId.intValue());`